### PR TITLE
demo: add a script

### PR DESCRIPTION
**Pull request by nicks**
Hello @jazzdan, @yuindustries,

Please review the following commits I made in branch nicks/ch586/checkout:

554877407d5f6c60ec101d57508e86e5436da17e (2018-10-17 20:04:12 -0400)
demo: add a script

ee3ddef9e82ac192a07dd686d72bdd424fc1e3f8 (2018-10-17 20:04:00 -0400)
demo: make the demo connect to the hud directly

6ab36f26c6af4a2c657f8c5353f2f870055681e1 (2018-10-17 20:04:00 -0400)
tiltfile: read relative to tiltfile dir instead of working dir
We expect that the Tilt demo will have an arbitrary workdir

7b5b823c90383e74e1f7a5e371b6db95c4688468 (2018-10-17 20:00:32 -0400)
build: fmt.Printf interrupts TTY / HUD. everything needs to go thru logger

### CODE REVIEW

**Comment by nicks** [comment]
this builds on a bunch of other PRs, but you can see the screencast here
https://asciinema.org/a/MDuD4Uufo4oNaXpZgK2smR5d1

File: internal/demo/script.go [hunk 1]
```diff
@@ -3,28 +3,183 @@ package demo
...
+	return s
+}
+
+type podMonitor struct {
+	podsReady   bool
+	podsReadyCh chan bool
+}
+
+func (m *podMonitor) OnChange(ctx context.Context, store *store.Store) {
+	state := store.RLockState()
```
Comment (jazzdan) [inline internal/demo/script.go]: Why isn't the defer for the unlock right after this call?

File: internal/demo/script.go [hunk 2]
```diff
@@ -3,28 +3,183 @@ package demo
...
+				return nil
+			}
+		}
 	}
 }
 
 func (s Script) Run(ctx context.Context) error {
-	fmt.Println("TODO(nick): implement demo")
+	if !s.env.IsLocalCluster() {
+		_, _ = fmt.Fprintf(os.Stderr, "tilt demo mode only supports docker-for-desktop or Minikube\n")
```
Comment (jazzdan) [inline internal/demo/script.go]: Maybe some additional verbiage here for docker-for-desktop? It seems less obvious and google-eable than Minikube

Comment (nicks) [reply internal/demo/script.go]: good point! i changed it to Docker For Mac

File: internal/demo/script.go (hunk 1 above)
Comment (nicks) [reply internal/demo/script.go]: ah, i meant to move this into a helper, good catch!

**Pull request merged**